### PR TITLE
Test Body Key scenario for request body

### PR DESCRIPTION
**Pull request by samvaity**

### CODE REVIEW

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/Test.RecordEntries/request_body_with_content.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      "RequestUri": "https://REDACTED/operatorInformation/:search?api-version=2024-03-01-preview",
+      "RequestMethod": "POST",
+      "RequestHeaders": {
+        "Accept": "application/json",
+        "Authorization": "Sanitized",
+        "Content-Length": "86",
+        "Content-Type": "application/json",
+        "User-Agent": "azsdk-java-azure-communication-phonenumbers/1.2.0-beta.2 (20.0.1; Windows 11; 10.0)",
+        "x-ms-client-request-id": "fafb6d96-04b5-4865-8cf8-0267c6e5bc6c",
+        "x-ms-date": "Tue, 12 Mar 2024 17:05:59 GMT",
```
Comment (scbedd) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/Test.RecordEntries/request_body_with_content.json:13]:
```suggestion
        "x-ms-date": "Tue, 12 Mar 2024 17:05:59 GMT"
```

Comment (scbedd) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/Test.RecordEntries/request_body_with_content.json]: This is the source.

**Comment by samvaity** [comment]
Closing as it works as expected.

**Pull request closed without merging**